src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java

```diff
@@ -312,6 +391,205 @@ public void processElement(final ProcessContext context) throws Exception {
+        }
+
+        @Override
+        public DuplicationMetrics decode( InputStream inStream, Context context ) throws CoderException, IOException {
+            DuplicationMetrics metrics = new DuplicationMetrics();
+            metrics.LIBRARY = SerializableCoder.of(String.class).decode(inStream, context);
+            metrics.UNMAPPED_READS = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.UNPAIRED_READS_EXAMINED = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.READ_PAIRS_EXAMINED = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.UNPAIRED_READ_DUPLICATES = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.READ_PAIR_DUPLICATES = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.READ_PAIR_OPTICAL_DUPLICATES = SerializableCoder.of(Long.class).decode(inStream, context);
+            metrics.PERCENT_DUPLICATION = SerializableCoder.of(Double.class).decode(inStream, context);
+            metrics.ESTIMATED_LIBRARY_SIZE = SerializableCoder.of(Long.class).decode(inStream, context);
+            return metrics;
+        }
+    }
+
+    private static void writeMetricsToFile(Pipeline pipeline, final PCollection<KV<String,DuplicationMetrics>> metrics,
+                                           final SAMFileHeader header, final File dest) {
+        PCollectionView<Map<String,Iterable<DuplicationMetrics>>> mapView = metrics
+            .apply(View.asMap());
+
+        PCollection<File> dummy = pipeline.apply(Create.<File>of(dest).setName("output metrics file"));
+
+        dummy.apply(ParDo.named("save metrics file")
+                    .withSideInputs(mapView)
+                    .of(new WriteToMetricsFile(header, mapView))
+        );
+
+    }
+
+    private static class WriteToMetricsFile extends DoFn<File,Void> implements Serializable {
+        private static final long serialVersionUID = 1L;
+        private final SAMFileHeader header;
+        private final PCollectionView<Map<String,Iterable<DuplicationMetrics>>> mapView;
+
+        public WriteToMetricsFile(SAMFileHeader header, PCollectionView<Map<String,Iterable<DuplicationMetrics>>> mapView) {
+            this.header = header;
+            this.mapView = mapView;
+        }
+
+        @Override
+        public void processElement(ProcessContext c) throws Exception {
+            File dest = c.element();
+            Map<String,Iterable<DuplicationMetrics>> metrics = c.sideInput(mapView);
+            final MetricsFile<DuplicationMetrics,Double> file = new MetricsFile<>();
+            for (final Map.Entry<String,Iterable<DuplicationMetrics>> entry : metrics.entrySet()) {
+                for (final DuplicationMetrics m : entry.getValue()) {
+                    file.addMetric(m);
+                }
+            }
+            // Should we add a histogram?
```

Comment:
The headers are just a time stamp and the input command line.  But they're formatted in a specific way

```
## htsjdk.samtools.metrics.StringHeader
# org.broadinstitute.hellbender.tools.picard.sam.markduplicates.MarkDuplicates  --INPUT /Users/louisb/Workspace/hellbender/src/test/resources/org/broadinstitute/hellbender/tools/picard/sam/MarkDuplicates/merge1.sam --INPUT /Users/louisb/Workspace/hellbender/src/test/resources/org/broadinstitute/hellbender/tools/picard/sam/MarkDuplicates/merge2.sam --INPUT /Users/louisb/Workspace/hellbender/src/test/resources/org/broadinstitute/hellbender/tools/picard/sam/MarkDuplicates/merge3.sam --OUTPUT /var/folders/xt/vq7wz8955r1401mv8w0f4zf9qbfwzl/T/MarkDuplicates.7414082829862458425.tmp/MarkDuplicates.sam --METRICS_FILE /var/folders/xt/vq7wz8955r1401mv8w0f4zf9qbfwzl/T/MarkDuplicates.7414082829862458425.tmp/MarkDuplicates.duplicate_metrics  --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --reportMemoryStats false --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --REMOVE_DUPLICATES false --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --READ_NAME_REGEX [a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).* --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --help false --version false --VERBOSITY INFO --QUIET false
## htsjdk.samtools.metrics.StringHeader
# Started on: August 3, 2015 2:33:08 PM EDT
```